encoding/src/parser.rs

```diff
@@ -970,6 +975,14 @@ impl<'a> FHDR<'a> {
 pub struct FCtrl(u8, bool);
 
 impl FCtrl {
+    pub fn new_uplink() -> FCtrl {
+        FCtrl(0x0, true)
+    }
+
+    pub fn set_ack(&mut self) {
```

Comment:
While this is a tiny type, so I can see the logic in suggesting to have this method here, but still I would rather have a separate builder for FCtrl and have that one used in `device/src/state_machines/session.rs` as it will be more consistent and easier to use when we need to set multiple properties of the type.